src/runtime/current_thread/runtime.rs

```diff
@@ -48,6 +50,21 @@ pub struct RunError {
     inner: current_thread::RunError,
 }
 
+impl fmt::Display for RunError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.inner)
+    }
+}
+
+impl Error for RunError {
+    fn description(&self) -> &str {
+        self.inner.description()
+    }
+    fn cause(&self) -> Option<&Error> {
+        self.inner.cause()
```

Comment:
Are you sure this should be `self.inner.cause()` and not just `self.inner`?

Reply:
I wasn't sure. But I felt this was actually the same error, just dressed in other clothes than one error causing another ‒ it wasn't like „Permission denied → Can't load configuration“ relation.